Code/Tools/AssetProcessor/native/tests/assetmanager/AssetProcessorManagerTest.cpp

```diff
@@ -4140,11 +4140,19 @@ struct LockedFileTest
         switch (message.GetMessageType())
         {
         case SourceFileNotificationMessage::MessageType:
-            if (const auto sourceFileMessage = azrtti_cast<const SourceFileNotificationMessage*>(&message);
-                sourceFileMessage != nullptr && sourceFileMessage->m_type == SourceFileNotificationMessage::NotificationType::FileRemoved
-                && m_callback)
+            if (const auto sourceFileMessage = azrtti_cast<const SourceFileNotificationMessage*>(&message); sourceFileMessage != nullptr &&
+                sourceFileMessage->m_type == SourceFileNotificationMessage::NotificationType::FileRemoved)
             {
-                m_callback();
+                // The File Remove message will occur before an attempt to delete the file
+                // Wait for more than 1 File Remove message.
+                // This indicates the AP has attempted to delete the file once, failed to do so and is now retrying
+                ++m_deleteCounter;
```

Comment:
why even add a delete counter to this class.
You could just add it to the callback

Reply:
like the caller could just have local var:
```
int deleteCounter = 0;
m_callBack = [&]() { 
    ++deleteCounter;
    if (deleteCounter > 1) 
     {
         product.close();
      }
   };
```

Reply:
Actually there is a trick where deleteCounter can be a member `int` of the callback itself, so that it doesn't have to be bound by reference

```
m_callBack = [&, deleteCounter = 0]() mutable { 
    ++deleteCounter;
    if (deleteCounter > 1) 
     {
         product.close();
      }
   };
```